test.disabled/ejabberd_tests/tests/mam_SUITE.erl

```diff
@@ -2538,6 +2571,191 @@ pagination_last_after_id5_before_id11(Config) ->
         end,
     parallel_story(Config, [{alice, 1}], F).
 
+
+%% Test cases for "complete" attribute
+%% Complete attribute is used for pagination, telling when to stop paginating.
+%% see complete_flag_cases with the whole list of the cases.
+%% -----------------------------------------------
+
+%% Get last page with most recent messages
+%% rsm_id.id is undefined
+%% GIVEN 15 archived messages
+%% WHEN direction=before, page_size=5
+%% THAN complete=false
+before_complete_false_last5(Config) ->
+    P = ?config(props, Config),
+    F = fun(Alice) ->
+        %% Get the last page of size 5.
+        %% Get messages: 11,12,13,14,15
+        RSM = #rsm_in{max=5, direction=before},
+        rsm_send(Config, Alice,
+            stanza_page_archive_request(P, <<"last5">>, RSM)),
+               wait_for_complete_respond(P, Alice, <<"false">>)
+        end,
+    parallel_story(Config, [{alice, 1}], F).
+
+%% Gets some page in the midle of the result set
+%% GIVEN 15 archived messages
+%% WHEN direction=before, rsm_id=10, page_size=5
+%% THAN complete=false
+before_complete_false_before10(Config) ->
+    P = ?config(props, Config),
+    F = fun(Alice) ->
+        %% Get messages: 5,6,7,8,9
+        RSM = #rsm_in{max=5, direction=before, id=message_id(10, Config)},
+        rsm_send(Config, Alice,
+            stanza_page_archive_request(P, <<"before10">>, RSM)),
+        wait_for_complete_respond(P, Alice, <<"false">>)
+        end,
+    parallel_story(Config, [{alice, 1}], F).
+
+%% Reaches the end of result set
+%% No messages are returned.
+%% GIVEN 15 archived messages
+%% WHEN direction=before, rsm_id=1, page_size=5
+%% THAN complete=true
+before_complete_true_before1(Config) ->
+    P = ?config(props, Config),
+    F = fun(Alice) ->
+        %% Get no messages
+        RSM = #rsm_in{max=5, direction=before, id=message_id(1, Config)},
+        rsm_send(Config, Alice,
+            stanza_page_archive_request(P, <<"before1">>, RSM)),
+        wait_for_complete_respond(P, Alice, <<"true">>)
+        end,
+    parallel_story(Config, [{alice, 1}], F).
+
+%% Reaches the end of result set
+%% GIVEN 15 archived messages
+%% WHEN direction=before, rsm_id=2, page_size=5
+%% THAN complete=true
+before_complete_true_before2(Config) ->
```

Comment:
Is this test case that much different from `before_complete_true_before5`? Does it test different branch of execution?

Reply:
I will remove this test.